### PR TITLE
feat(JWTToken): implement JWTToken functionality

### PR DESCRIPTION
**Pull request by davidshare**
#### What does this PR do?
Add token generation and authentication functionality
#### Description of Task to be completed?
- install babel-plugin-transform-object-rest-spread
- add babel-plugin-transform-object-rest-spread as a plugin to the .babelrc
- add a secret key to the .env file for encrypting tokens
- create a JWTHelper class for handling token manipulation
- create a static method to the JWTHelper for generating user tokens
- create a static method to the JWTHelper for verifying and decoding user tokens
- create a middleware for verifying the presence of tokens in the user object
- install faker and rosie for mocking use data
- create mock user data for testing functions
- create a function for stripping the user object of the password and the timestamps
#### How should this be manually tested?
- Clone the repo. E.g on the commandline, type git clone https://github.com/andela/selene-ah-backend.git
- type: npm install
- start the app with the following command: npm start
- signup as a user
- after a successful signup, you will be given a token
#### Any background context you want to provide?
- none
#### What are the relevant pivotal tracker stories?
[#161779861](https://www.pivotaltracker.com/story/show/161779861)
#### Screenshots (if appropriate)
- none
#### Questions:
- none

### CODE REVIEW

File: server/helpers/JWTHelper.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+class JWTHelper {
+  /**
+   * @description This function generates JWT tokens
+   * @param {object} userObject
+   * @returns {string} token
+   */
+  static generateToken(userObject){
+    if(!userObject || !userObject.firstName) {
+      return 'Please supply a valid user object.';
```
Comment (theghostyced) [inline server/helpers/JWTHelper.js]: Nice work @davidshare. But we could throw an error here

File: server/helpers/JWTHelper.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  /**
+   * @description This function verifies and decodes JWT tokens
+   * @param {string} userToken
+   * @returns {Object} userObject
+   */
+  static verifyToken(userToken){
+    if(!userToken || typeof userToken !=='string'){
+      return 'Please enter a valid token.';
```
Comment (theghostyced) [inline server/helpers/JWTHelper.js]: Also here we could throw an error here too.

File: server/middlewares/authentication.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+class Authentication{
+
+  /**
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {function} next
+   */
+  static authenticateUser(req, res, next) {
+    try {
```
Comment (theghostyced) [inline server/middlewares/authentication.js]: Lovely work here. But could you please explain why you have a `try..catch` block

File: server/helpers/JWTHelper.js (hunk 2 above)
Comment (davidshare) [reply server/helpers/JWTHelper.js]: Done

File: server/middlewares/authentication.js (hunk 3 above)
Comment (davidshare) [reply server/middlewares/authentication.js]: An error is being thrown from the JWT.verifyToken method.

**Review by Danielshow — APPROVED** [review]
LGTM

**Review by ekpangmichael — APPROVED** [review]
Looks good to me

**Review by efalayi — APPROVED** [review]

**Pull request merged**